src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +32,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String splitAddress[] = trimmedAddress.split(ADDRESS_SPLITTER);
+        if (splitAddress.length != ADDRESS_PARAMETERS){
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+        block = new Block(splitAddress[0]);
+        street = new Street(splitAddress[1]);
+        unit = new Unit(splitAddress[2]);
+        postalCode = new PostalCode(splitAddress[3]);
+        this.value = block.getBlockNumber() + ADDRESS_SPLITTER + street.getStreetName() + ADDRESS_SPLITTER + unit.getUnitNumber() + ADDRESS_SPLITTER + postalCode.getPostalCode();
```

Comment:
Do you think that there is a need to retain the whole address value? Can you simply reconstruct the value when needed?